cypher-shell/src/test/java/org/neo4j/shell/MainTest.java

```diff
@@ -171,6 +171,27 @@ public void connectInteractivelyTriesOnlyOnceIfUserPassExists() throws Exception
     }
 
     @Test
+    public void connectInteractivelyRepromptsIfUserIsNotProvided() throws Exception {
+        doThrow(authException).doNothing().when(shell).connect(connectionConfig);
+
+        String inputString = "\nbob\nsecret\n";
+        InputStream inputStream = new ByteArrayInputStream(inputString.getBytes());
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        PrintStream ps = new PrintStream(baos);
+
+        Main main = new Main(inputStream, ps);
+        main.connectInteractively(shell, connectionConfig);
+
+        String out = new String(baos.toByteArray(), StandardCharsets.UTF_8);
+
+        assertEquals(out, "username: \r\nusername cannot be empty\n\nusername: bob\r\npassword: ******\r\n");
```

Comment:
Your test doesn't include the extra newline which the bare `out.println()` gives in the method.